ui-v2/app/adapters/service.js

```diff
@@ -1,15 +1,24 @@
 import Adapter from './application';
 import { PRIMARY_KEY, SLUG_KEY } from 'consul-ui/models/service';
 import { OK as HTTP_OK } from 'consul-ui/utils/http/status';
+const URL_PREFIX_SINGLE = 'health/service';
+const URL_PREFIX_MULTIPLE = 'internal/ui/services';
 export default Adapter.extend({
   urlForQuery: function(query, modelName) {
-    return this.appendURL('internal/ui/services', [], this.cleanQuery(query));
+    return this.appendURL(URL_PREFIX_MULTIPLE, [], this.cleanQuery(query));
   },
   urlForQueryRecord: function(query, modelName) {
     if (typeof query.id === 'undefined') {
       throw new Error('You must specify an id');
     }
-    return this.appendURL('health/service', [query.id], this.cleanQuery(query));
+    return this.appendURL(URL_PREFIX_SINGLE, [query.id], this.cleanQuery(query));
+  },
+  isQueryRecord: function(url, method) {
```

Comment:
Are you overriding an adapter method here? Curious why this new function wasn't called and couldn't find it in the adapter source after a quick look.

Reply:
Yeah it overrides a 'generic' method in the ApplicationAdapter. The ApplicationAdapter has a generic assumption for knowing whether a URL was to query a record (as oppose to query record**s**). This Adapter inherits from that and therefore previously used the 'generic' `isQueryRecord`, using this 'generic' method was the root of this bug.

The 'generic' assumption was that the service name was always the last 'part' of the URL (so `/health/service/service-name`). That was the wrong assumption to make :) and the root of this bug, as people could have services with slashes in them meaning the last 'part' is no longer the full service name.

> Curious why this new function wasn't called and couldn't find it in the adapter source after a quick look.

It's down in the `handleResponse` method

https://github.com/hashicorp/consul/blob/d8c9ab318543dd79aa3c2fa112b1a88adb0157a6/ui-v2/app/adapters/service.js#L23-L37